src/test/java/io/github/ust/mico/kafkafaasconnector/MessageListenerTests.java

```diff
@@ -34,24 +52,85 @@
 import io.github.ust.mico.kafkafaasconnector.kafka.MicoCloudEventImpl;
 import io.github.ust.mico.kafkafaasconnector.kafka.RouteHistory;
 
+import javax.annotation.PostConstruct;
+
+import static org.hamcrest.CoreMatchers.equalTo;
+import static org.hamcrest.core.IsNull.notNullValue;
+import static org.junit.Assert.*;
+import static org.hamcrest.Matchers.is;
+import static org.hamcrest.Matchers.empty;
+
 @RunWith(SpringRunner.class)
 @SpringBootTest(webEnvironment = SpringBootTest.WebEnvironment.MOCK)
 @EnableAutoConfiguration
 @AutoConfigureMockMvc
 @ActiveProfiles("testing")
 @DirtiesContext
+@Slf4j
 public class MessageListenerTests {
 
+    @Autowired
+    private KafkaConfig kafkaConfig;
+
+    private static boolean kafkaTopicInitDone = false;
+
     // https://docs.spring.io/spring-kafka/docs/2.2.6.RELEASE/reference/html/#kafka-testing-junit4-class-rule
     @ClassRule
-    public static EmbeddedKafkaRule broker = new EmbeddedKafkaRule(1, false,
-        TestConstants.INPUT_TOPIC, TestConstants.OUTPUT_TOPIC,
-        TestConstants.DEAD_LETTER_TOPIC, TestConstants.INVALID_MESSAGE_TOPIC,
-        TestConstants.ROUTING_TOPIC_1, TestConstants.ROUTING_TOPIC_2,
-        TestConstants.ROUTING_TOPIC_3, TestConstants.ROUTING_TOPIC_4);
+    public static EmbeddedKafkaRule broker = new EmbeddedKafkaRule(1, false);
 
     @Autowired
-    private MessageListener messageListener;
+    MessageListener messageListener;
+
+    @PostConstruct
+    public void before(){
+        //We need to add them outside of the rule because the autowired kakfaConfig is not accessible from the static rule
+        //We can not use @BeforeClass which is only executed once because it has to be static and we do not have access to the autowired kakfaConfig
+
+        Set<String> requiredTopics = getRequiredTopics();
```

Comment:
Forgot to use kafkaTopicInitDone to not execute this for each test?

Reply:
No, I tried different approaches. One was based on @Before and there I needed kafkaTopicInitDone. I will remove it.